test/e2e/yurt/iot.go

```diff
@@ -0,0 +1,142 @@
+		By("make sure the needed namespace are removed")
+
+		res := &corev1.Namespace{}
+		Eventually(
+			func() error {
+				return k8sClient.Get(ctx, client.ObjectKey{
+					Name: namespaceName,
+				}, res)
+			}).WithTimeout(timeout).WithPolling(time.Millisecond * 500).Should(&util.NotFoundMatcher{})
+		Eventually(
+			func() error {
+				return k8sClient.Create(ctx, &ns)
+			}).WithTimeout(timeout).WithPolling(time.Millisecond * 300).Should(SatisfyAny(BeNil(), &util.AlreadyExistMatcher{}))
+	}
+
+	createPlatformAdmin := func(version string) {
+		Eventually(func() error {
+			return k8sClient.Delete(ctx, &iotv1alpha2.PlatformAdmin{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      platformAdminName,
+					Namespace: namespaceName,
+				},
+			})
+		}).WithTimeout(timeout).WithPolling(500 * time.Millisecond).Should(SatisfyAny(BeNil(), &util.NotFoundMatcher{}))
+
+		testPlatformAdmin := iotv1alpha2.PlatformAdmin{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      platformAdminName,
+				Namespace: namespaceName,
+			},
+			Spec: iotv1alpha2.PlatformAdminSpec{
+				Version:  version,
+				PoolName: nodePoolName,
+			},
+		}
+		Eventually(func() error {
+			return k8sClient.Create(ctx, &testPlatformAdmin)
+		}).WithTimeout(timeout).WithPolling(500 * time.Millisecond).Should(SatisfyAny(BeNil(), &util.AlreadyExistMatcher{}))
+	}
+
+	BeforeEach(func() {
+		By("Start to run iot test, clean up previous resources")
+		k8sClient = ycfg.YurtE2eCfg.RuntimeClient
+		createNamespace()
+	})
+
+	AfterEach(func() {
+		By("Cleanup resources after test")
+		By(fmt.Sprintf("Delete the entire namespaceName %s", namespaceName))
+		Expect(k8sClient.Delete(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}}, client.PropagationPolicy(metav1.DeletePropagationBackground))).Should(BeNil())
+	})
+
+	for _, testVersion := range testVersions {
+		version := testVersion
+		Describe(fmt.Sprintf("Test the %s version of PlatformAdmin", version), func() {
+			BeforeEach(func() {
+				platformAdminName = "test-platform-admin-" + version
+				createPlatformAdmin(version)
+			})
+
+			AfterEach(func() {
+				By(fmt.Sprintf("Delete the platformAdmin %s", platformAdminName))
+				Expect(k8sClient.Delete(ctx, &iotv1alpha2.PlatformAdmin{ObjectMeta: metav1.ObjectMeta{Name: platformAdminName, Namespace: namespaceName}}, client.PropagationPolicy(metav1.DeletePropagationBackground))).Should(BeNil())
+			})
+
+			It(fmt.Sprintf("The %s version of PlatformAdmin should be stable in ready state after it is created", version), func() {
+				By("verify the status of platformadmin")
+				Eventually(func() error {
+					testPlatfromAdmin := &iotv1alpha2.PlatformAdmin{}
+					if err := k8sClient.Get(ctx, types.NamespacedName{Name: platformAdminName, Namespace: namespaceName}, testPlatfromAdmin); err != nil {
+						return err
+					}
+					if testPlatfromAdmin.Status.Ready == true {
```

Comment:
Platformadmin deploys required components by default, but just platformadmin is ready does not mean that required components are ready.

Reply:
Thank you for reminding me. I don't understand what this means, but the Ready should determine the state of the component (default, optional, custom).
https://github.com/openyurtio/openyurt/blob/068fefe9e7b22d47d2dd5e13d4dbf7953f01e60d/pkg/yurtmanager/controller/platformadmin/platformadmin_controller.go#L303-L335
Here the component is synchronized from the framework.
https://github.com/openyurtio/openyurt/blob/068fefe9e7b22d47d2dd5e13d4dbf7953f01e60d/pkg/yurtmanager/controller/platformadmin/platformadmin_controller.go#L501C2-L501C2

Reply:
What I mean is that I'm aware of the fact that platformadmin is truly ready in the sense that all components under platformadmin are ready.
https://github.com/openyurtio/openyurt/blob/068fefe9e7b22d47d2dd5e13d4dbf7953f01e60d/pkg/yurtmanager/controller/platformadmin/platformadmin_controller.go#L321-L330
we use `ReadyComponentNum` and `UnreadyComponentNum` to indicate the number of ready and unready components respectively. Is it possible to use this as a way of evaluating the real situation of components under platformadmin.
https://github.com/openyurtio/openyurt/blob/068fefe9e7b22d47d2dd5e13d4dbf7953f01e60d/pkg/yurtmanager/controller/platformadmin/platformadmin_controller.go#L376-L396

Reply:
I got it. Thank you for pointing it out.